packages/grid/_modules_/grid/hooks/features/keyboard/useGridKeyboardNavigation.ts

```diff
@@ -137,13 +164,78 @@ export const useGridKeyboardNavigation = (
     ],
   );
 
+  const navigateColumnHeders = React.useCallback(
+    (params: GridCellParams, event: React.KeyboardEvent) => {
+      event.preventDefault();
+      let nextColumnHeaderIndexes: GridColumnHeaderIndexCoordinates | null;
+      const { colIndex } = params;
+      const key = mapKey(event);
+      const ColumnHeaderEl = params.element!;
+      ColumnHeaderEl.tabIndex = -1;
+
+      if (isArrowKeys(key)) {
+        nextColumnHeaderIndexes = getNextColumnHeaderIndexes(key, {
+          colIndex,
+        });
+      } else if (isHomeOrEndKeys(key)) {
+        const colIdx = key === 'Home' ? 0 : colCount - 1;
+
+        nextColumnHeaderIndexes = { colIndex: colIdx };
+      } else if (isPageKeys(key)) {
+        // Handle only Page Down key, Page Up should keep the current possition
+        if (key.indexOf('Down') > -1) {
+          apiRef.current.setCellFocus({
+            colIndex: params.colIndex,
+            rowIndex: containerSizes!.viewportPageSize - 1,
```

Comment:
Why move the focus back to the last but one visible row? Should not be the row that had focus before <kbd>PageUp</kbd> or fallback to the first?

Reply:
Because that is what the bottom of the page is. An interaction like this is not really logical at least IMO.

Reply:
I asked because I found this a bit weird.

https://user-images.githubusercontent.com/42154031/113713326-93f19300-96bd-11eb-9a07-31821218af79.mp4